### ISSUE TITLE
Be able to create an indicator in cluster reporting for a cluster objective, cluster activity and partner project.

### ISSUE THREAD

**Issue by chachra** [unicef/etools-partner-reporting-portal#133]
We will tackle edit in a separate task I feel since there will be restrictions etc. on it. Please ignore indicators associated with a partner activity for this task as well.

Do after #166 is in good place in a feature branch or develop branch.

Cluster reporting PRP mockup is here: https://projects.invisionapp.com/share/EDBIM0Y8M#/screens/232533792

The frontend can send N disaggregation each with at least once choice. The disaggregation should always have a name.

Any disaggregation removed (not sent by frontend?) can be marked as inactive then. same for disaggregation choice removed.

**Technical Design**

Please build on branch for #166 if not merged to develop.

Use chips: https://material.io/guidelines/components/chips.html

API needs to support GET for reading the data and POST for creation and editing as well.

For frontend it will be helpful to see the existing code in PMP and can take components or at least layout ideas from there: https://github.com/unicef/etools-partnership-management/blob/master/pmp/app-elements/interventions/expected-results/indicator-dialog.html

**Acceptance criteria**
* User creates a cluster objective
* Goes to objective overview, clicks on indicators
* doesn't see any indicators, clicks create indicators
* all required data is entered, validations above work fine
* After addition, indicator appears in the indicator list (if indicator list task is done before this feature or at least in the db in a valid form with all its disaggregations).

**Comment by lkolacz** [comment]
btw.. branch name `feature-create-edit-indicator-#133`

**Comment by kasialanecka** [comment]
@chachra I think I have the changes captured, please check.

**Comment by chachra** [comment]
@lkolacz please review the mockup and make sure API is in place to make this happen. @n1207n has been working on some data model changes to support this as well, so see his PR. Not super urgent.

**Comment by lkolacz** [comment]
after review, this issue need update!

**Comment by lkolacz** [comment]
updates implemented after mockup changes -> API in place!

**Comment by chachra** [comment]
@lkolacz API doesn't support creation of disaggregations? Please fix this asap... also keep #168 in mind... so edit doesn't take too long to implement. API design should enable edit capability in the future easily. Tx.

**Comment by chachra** [comment]
@rkrupinski assigned to you, please review and ask any functional questions etc. please. @lkolacz any response to above comment from last week? Please come to this task when done with others today to verify indicator creation is in place. Thanks.

**Comment by chachra** [comment]
@lkolacz please merge develop to this branch.

**Comment by lkolacz** [comment]
@chachra DONE!

**Comment by lkolacz** [comment]
how to post to endpoint .. example...

```
{
            'object_id': 1,
            'object_type': 'ClusterObjective',
            'means_of_verification': 'IMO/CC calculation',
            'locations': [
                {'id': 1},
                {'id': 2},
            ],
            'blueprint': {
                'title': 'of temporary classrooms',
                'unit': 'number',
                'description': 'Average measure for the month',
                'calculation_formula_across_periods': 'max',
                'calculation_formula_across_locations': 'avg',
                'display_type': 'number',
                'disaggregatable': True,
            },
            'disaggregation': [
                {'name': 'Age', 'values': ['0-5m', '5-12m', '1-3y', '4-7y']},
                {'name': 'Gender', 'values': ['Male', 'Female', 'Other']}
            ]
        }
```

**Comment by lkolacz** [comment]
```
{'blueprint': {'calculation_formula_across_locations': u'avg', 'description': 'Average measure for the month', 'title': 'of temporary classrooms', 'calculation_formula_across_periods': u'max', 'disaggregatable': True, 'unit': u'number', 'display_type': u'number'}, 'means_of_verification': 'IMO/CC calculation', 'object_type': 'ClusterActivity', 'disaggregation': [{'values': ['0-5m', '5-12m', '1-3y', '4-7y'], 'name': 'Age'}, {'values': ['Male', 'Female', 'Other'], 'name': 'Gender'}], 'locations': [{'id': 116}, {'id': 128}], 'object_id': 55, 'frequency': u'Csd', 'cs_dates': ['2017-07-27', '2017-07-30', '2017-08-02', '2017-08-05']}
```

**Comment by rkrupinski** [comment]
@lkolacz Please limit the length of disaggregation values to 15 characters.

**Comment by lkolacz** [comment]
@rkrupinski OK

**Comment by rkrupinski** [comment]
@ErinOC Here's how you add indicators using the `<indicator-modal>` component:

https://github.com/unicef/etools-partner-reporting-portal/blob/4ca0a9495da8fe2a1767b5553a4df78d6f301910/polymer/src/pages/app/cluster-reporting/analysis.html#L29
